### PR TITLE
Fix baby animal sizes

### PR DESCRIPTION
**Pull request by OnlyBMan**
This should fix most baby animal sizes including Rabbits and their adult sizes!

### CODE REVIEW

**Comment by EOT3000** [comment]
what about cats? dogs? zombies?

**Comment by RednedEpic** [comment]
> what about cats? dogs? zombies?

Cats and dogs extend AgeableEntity. As for zombies, it appears that is missing. However, it should be an easy fix.

File: connector/src/main/java/org/geysermc/connector/entity/living/RabbitEntity.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+            metadata.getFlags().setFlag(EntityFlag.BABY, isBaby);
+            if(isBaby) {
+                metadata.put(EntityData.SCALE, .35f);
+            }
+            else {
+                metadata.put(EntityData.SCALE, .55f);
+            }
+        }
+
+        super.updateBedrockMetadata(entityMetadata, session);
```
Comment (RednedEpic) [inline connector/src/main/java/org/geysermc/connector/entity/living/RabbitEntity.java]: Couldn't you call `super.updateBedrockMetadata` first, then remove the check altogether in `AgeableEntity` for rabbits?

File: connector/src/main/java/org/geysermc/connector/entity/living/AgeableEntity.java [hunk 2]
```diff
@@ -40,8 +41,12 @@ public AgeableEntity(long entityId, long geyserId, EntityType entityType, Vector
...
     @Override
     public void updateBedrockMetadata(EntityMetadata entityMetadata, GeyserSession session) {
-        if (entityMetadata.getId() == 15) {
-            metadata.getFlags().setFlag(EntityFlag.BABY, (boolean) entityMetadata.getValue());
+        if (entityMetadata.getId() == 15 && entityType != EntityType.RABBIT) {
+            boolean isBaby = (boolean) entityMetadata.getValue();
+            metadata.getFlags().setFlag(EntityFlag.BABY, isBaby);
+            if (isBaby) {
+                metadata.put(EntityData.SCALE, .55f);
+            }
```
Comment (RednedEpic) [inline connector/src/main/java/org/geysermc/connector/entity/living/AgeableEntity.java]: See comment below.

**Comment by EOT3000** [comment]
woah

**Comment by RednedEpic** [comment]
👍

**Pull request merged**